### DISCUSSION TITLE
Extract Method Code Action

### DISCUSSION THREAD

**Discussion by arrrrny** [zed-industries/zed#26080] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

One of the features I used to use very often was Extract Method code action. When I see a duplication of Code in a few places, I simply highlight the code and  click 'Extract Method' code action and now I have a reusable internal method. I used this  a lot on my vsc days. This is great for 'refactoring'

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="658" alt="Screenshot 2024-08-13 at 10 47 39 AM" src="https://github.com/user-attachments/assets/883cd4d7-cf64-4dc3-8f30-8af627763275">

![Screenshot 2024-08-13 at 10 48 13 AM](https://github.com/user-attachments/assets/fd321675-eea5-467e-adf7-d3b74e5111eb)

![Screenshot 2024-08-13 at 10 49 03 AM](https://github.com/user-attachments/assets/bd0eb204-e2dc-47f6-a21c-a3fa7a8f6a3d)

**Comment by michalmela** [comment]
Jetbrains IDEs also have _extract variable_ and _extract constant_ refactorings. While not as important as extracting functions, they would also be nice to have, so putting it out there.
